physioqc/interfaces/visualizations.py

```diff
@@ -0,0 +1,197 @@
+    figsize : tuple, optional
+        Size of the new figure (if created), by default (10, 5)
+
+    Returns
+    -------
+    Tuple[plt.figure, plt.axes]
+        Figure and axes objects of the plot.
+    """
+
+    if ax is None:
+        fig, ax = plt.subplots(1, 1, figsize=figsize)
+    else:
+        fig = None
+
+    return fig, ax
+
+
+def plot_raw_data(phys: pk.Physio, ax: plt.axes = None) -> Tuple[plt.figure, plt.axes]:
+    """Plots the raw data using peakdet.
+    Parameters
+    ----------
+    phys : pk.Physio
+        The data object, used for plotting.
+    ax : plt.axes, optional
+        axes object to use for plotting, by default None
+
+    Returns
+    -------
+    Tuple[plt.figure, plt.axes]
+        Figure and axes objects of the plot.
+    """
+    fig, ax = check_create_figure(ax, figsize=(7, 5))
+
+    ax = pk.plot_physio(phys, ax=ax)
+
+    return fig, ax
+
+
+def plot_average_peak(
+    phys: pk.Physio,
+    window: List = [-3, 3],
+    target_fs: float = 1000.0,
+    peak_dist: float = 1.0,
+    peak_thr: float = 0.1,
+    plot_mode: Literal["traces", "ci", "auto"] = "auto",
+    ax: plt.axes = None,
+) -> Tuple[plt.figure, plt.axes]:
+    """_summary_
+
+    Parameters
+    ----------
+    phys : pk.Physio
+        data object, used for plotting.
+    window : List, optional
+        window size around the peak in s, by default [-3, 3]
+    target_fs : float, optional
+        sampling rate for plotting and peak detection, by default 1000.0
+    peak_dist : float, optional
+        distance for peak detection, by default 1.0
+    peak_thr : float, optional
+        threshold for peak detection, by default 0.1
+    plot_mode : Literal["traces", "ci"], optional
+        to plot traces or standard deviation around signal, by default 'auto'
+    ax : plt.axes, optional
+        axes object to use for plotting, by default None
+
+
+    Returns
+    -------
+    Tuple[plt.figure, plt.axes]
+        Figure and axes objects of the plot.
+
+    Raises
+    ------
+    ValueError
+        If an inapropriate plot type is chosen.
+    """
+    if plot_mode not in ["traces", "ci", "auto"]:
+        raise ValueError('Plot mode has to be in ["traces", "ci", "auto"]')
+
+    if phys.fs > 1_000:
+        #  This should probably go to the logs, or warning
+        print(f"Fs = {phys.fs} Hz too high, resampling to {target_fs} Hz")
+        phys = pk.operations.interpolate_physio(phys, target_fs, kind="linear")
+
+    phys = multimodal.peak_detection(phys, peak_dist=peak_dist, peak_threshold=peak_thr)
+
+    window = np.array(window) * phys.fs
+    window = window.astype(int)
+
+    window_range = np.arange(window[0], window[1] + 1)
+    t = window_range / phys.fs
+
+    # Disregard not used peaks:
+    peaks = list(
+        filter(
+            lambda ps: ((ps + window[0]) >= 0)
+            and ((ps + window[1] + 1) < phys.data.shape[0]),
+            phys.peaks,
+        )
+    )
```

Comment:
This part was a bit confusing to me, why are we not using these peaks?

Reply:
If I remember correctly, it is discarding peaks where the data is not fully covered by the window. The idea of the figure is to plot the signal shape around peaks. (Hope that makes sense)

Reply:
That does, thanks! That's my understanding of what it's doing, but I'm wondering whether it wouldn't make more sense to include all peaks in the average, regardless of whether some data is being replicated inside the window.

Reply:
Not sure what you mean with "replicated". The filtering makes the process slightly easier, as it avoid indexing errors. Otherwise, some padding would be needed, but that complicates the timing (very slightly).

Reply:
Oh, I see, I misunderstood. That makes sense to me the way it is then, thanks! Maybe add a comment to say that this is just a question of indexing errors so some peaks at the beginning/end of the series may be excluded.